server/gitrest/src/routes/summaries.ts

```diff
@@ -392,12 +392,33 @@ export function create(
         const resultP = repoManagerFactory.open(
             request.params.owner,
             request.params.repo,
-        ).then(async (repoManager) => createSummary(
-            repoManager,
-            wholeSummaryPayload,
-            documentId,
-            getExternalWriterParams(request.query?.config as string | undefined)?.enabled ?? false,
-        ));
+        ).then(async (repoManager): Promise<IWriteSummaryResponse | IWholeFlatSummary> => {
+            const externalStorageEnabled =
+                getExternalWriterParams(request.query?.config as string | undefined)?.enabled ?? false;
+            const writeSummaryResponse = await createSummary(
+                repoManager,
+                wholeSummaryPayload,
+                documentId,
+                externalStorageEnabled,
+            );
+            if (wholeSummaryPayload.type === "container") {
+                try {
+                    const wholeFlatSummary = await getSummary(
+                        repoManager,
+                        writeSummaryResponse.id,
+                        documentId,
+                        externalStorageEnabled,
+                    );
+                    return wholeFlatSummary;
+                } catch (e) {
+                    // This read is for Historian caching purposes, so it should be ignored on failure.
+                    winston.error("Failed to read latest summary after writing container summary", {
+                        documentId,
+                    });
+                }
```

Comment:
Do we need to rethrow the error caught in line 413? Or does `handleResponse` understand a problem happened even if we do not rethrow the error here?

Reply:
My reasoning here was to intentionally ignore the error, because the read operation within the try..catch is only useful for caching within Historian, and has no retry mechanism. If the read fails, the response will simply return `{ id }` as it does today, and nothing will be cached in Historian, so the next Read from a client will actually hit storage.